tutorials/darcy_thermo_mech/doc/content/workshop/numerical/fem_overview.md

```diff
@@ -2,89 +2,224 @@
+- +Discrete sense+: Using sampled locations with associated values, we want our polynomial function to be as close to the given data as possible.
+- +Continuous sense+: Using a complicated function, we want to have our polynomial function as close to the given function as possible.
 
-where $a$, $b$ and $c$ are scalar coefficients and $1$, $x$, $x^2$ are "basis functions". Thus, the
-problem is to find $a$, $b$, $c$, etc. such that $f(x)$ passes through the points given.
 
-More generally,
+Let us take the polynomial in the following form:
 
 !equation
-f(x) = \sum_{i=0}^d c_i x^i,
+f(x) = \sum_{i=0}^d c_i x^i
 
-where the $c_i$ are coefficients to be determined. $f(x)$ is unique and interpolary if $d+1$ is the
-same as the number of points needed to fit. This defines a linear system that must be solved to
-find the coefficients.
+where $c_i$ are scalar coefficients (expansion coefficients) and monomials
+$x^i$ are "basis functions". Thus, the
+problem is to find $c_i$ such that $f(x)$ is closest the the given points or a
+given function.
 
 !---
 
-## Polynomial Example
+## Polynomial Example (Discrete)
+
+!row!
+!col! width=49%
 
-Define a set of points:
+Define a set of points (Let's pick the points using $sin(x)$):
 
 !equation
-(x_1, y_1) = (1,4) \\
-(x_2, y_2) = (3,1) \\
-(x_3, y_3) = (4,2)
+(x_1, y_1) = (0.25,0.247) \\
+(x_2, y_2) = (0.50,0.479) \\
+(x_3, y_3) = (0.75,0.682) \\
+(x_4, y_4) = (1.00,0.841)
+
 
-Substitute $(x_i, y_i)$ data into the model:
+Substitute $(x_i, y_i)$ data into the polynomial model:
 
 !equation
-y_i = a + bx_i + cx_i^2, i=1,2,3.
+c_0 + c_1 x_i + c_2 x_i^2 = y_i,\quad i=1,..,4.
 
-This leads to the following linear system for $a$, $b$, and $c$:
+This leads to the following linear system for $c_i$:
 
 !equation
+\mathbf{A}
+\vec{c}
+=
+\vec{y}
+
+where
+
+!style fontsize=80%
+!equation
+\mathbf{A} =
 \begin{bmatrix}
-  1 & 1 &  1 \\
-  1 & 3 &  9 \\
-  1 & 4 & 16
-\end{bmatrix}
-\begin{bmatrix}
-  a \\
-  b \\
-  c
+A_{*0} | A_{*1} | A_{*2}
 \end{bmatrix}
 =
 \begin{bmatrix}
-  4 \\
-  1 \\
-  2
+  1 & 0.25 & 0.25^2 \\
+  1 & 0.50 & 0.50^2 \\
+  1 & 0.75 & 0.75^2 \\
+  1 & 1.00 & 1.00^2
+\end{bmatrix}, ~~
+\vec{c} = \begin{bmatrix}
+  c_0 \\
+  c_1 \\
+  c_2
+\end{bmatrix}, ~~
+\vec{y} =
+\begin{bmatrix}
+  0.247 \\
+  0.479 \\
+  0.682 \\
+  0.841
 \end{bmatrix}
 
-!---
+!col-end!
 
-Solving for the coefficients results in:
+!col! width=49%
+
+!alert! note title=Solving this problem
+
+We want to get the function closest to the points.
+Distance in this case can be expressed by the discrete
+$L^2$ norm:
+
+!equation
+R = ||\mathbf{A}\vec{c}-\vec{y}||^2_2,\quad \text{where} \quad
+||\vec{v}||^2_2 = \vec{v}^T \vec{v},
+
+We want to minimize the norm of the squared distances ("least squares"):
+
+!equation
+\frac{\partial R}{\partial c_i} = 2 A_{*i}^T (\mathbf{A}\vec{c} - \vec{y}) = 0
+
+which requires the solution of this problem:
+
+!equation
+\mathbf{A}^T \mathbf{A}\vec{c} = \mathbf{A}^T\vec{y}
```

Comment:
are you introducing gradient descent to get here?

Reply:
like the intent is to say, "we'll do one step of gradient descent, and since this is linear, we'll be there" ?